dali/operators/reader/loader/video_loader.cc

```diff
@@ -132,33 +133,57 @@ vector<std::pair<string, int>> filesystem::get_file_label_pair(
     // could return directories with the same names in completely different order
     std::sort(entry_name_list.begin(), entry_name_list.end());
     for (unsigned dir_count = 0; dir_count < entry_name_list.size(); ++dir_count) {
-        assemble_video_list(file_root, entry_name_list[dir_count], dir_count, file_label_pairs);
+        assemble_video_list(file_root, entry_name_list[dir_count], dir_count, file_info);
     }
 
     // sort file names as well
-    std::sort(file_label_pairs.begin(), file_label_pairs.end());
+    std::sort(file_info.begin(), file_info.end());
   } else if (!file_list.empty()) {
     // load (path, label) pairs from list
     std::ifstream s(file_list);
     DALI_ENFORCE(s.is_open(), file_list + " could not be opened.");
 
+    string line;
     string video_file;
     int label;
-    while (s >> video_file >> label) {
-      file_label_pairs.push_back(std::make_pair(video_file, label));
+    float start_time;
+    float end_time;
+    int line_num = 0;
+    while (std::getline(s, line)) {
+      line_num++;
+      video_file.clear();
+      label = start_time = end_time = -1;
+      std::istringstream file_line(line);
+      file_line >> video_file >> label;
+      if (video_file.empty()) continue;
+      DALI_ENFORCE(label >= 0, "Label value should be >= 0 in file_list at line number: "
+                   + to_string(line_num) + ", filename: "+ video_file);
+      if (file_line >> start_time) {
+        DALI_ENFORCE(start_time >= 0, "Start time/frame should be >=0 at line number: "
+                     + to_string(line_num) + ", filename: "+ video_file);
+        if (file_line >> end_time) {
+          DALI_ENFORCE(start_time <= end_time, "Start time/frame should be <= end time/frame "
+                       "at line number: " + to_string(line_num) + ", filename: "+ video_file);
+        } else {
+          DALI_FAIL("Either omit start time/frame and end time/frame, or provide both at "
```

Comment:
It can fail also when instead of numeric value some char is provided. I would reword it to something like this:
```
The allowed file list format is:
file_name, label (int), start_time (float), end_time (float)
or 
file_name, label (int)
```